packages/mdc-chips/chip/index.js

```diff
@@ -59,6 +61,7 @@ class MDCChip extends MDCComponent {
   }
 
   initialize() {
+    this.id = this.root_.id || this.root_.innerText.slice(0, -1) + Math.random();
```

Comment:
`innerText` seems unwieldy for this, and `Math.random()` is going to produce incomprehensible decimal numbers. A better idea might be to use a module-local counter:

```js
let idCounter = 0;

...

initialize() {
  this.id = this.root_.id || `mdc-chip-${++idCounter}`;
```

Also, should we set the `id` attribute on the element too?

Tangentially, I'm not sure what the purpose of `slice(0, -1)` was (FYI `trim` should work in all supported browsers).

Reply:
To use a counter, the id will be set in the chip set. I used the id of the chip set to minimize id collisions in case there's more than one chip set. Do you think that would be secure enough?

(And thanks for the tip, `trim` was what I was looking for 🙂)

Reply:
I updated the README to recommend setting an id on the chip set

Reply:
If you initialize a counter on every chip set, yes, you'd have to do this, and as implemented, this will break if the user ever doesn't set a chip set ID, so you've effectively just punted the issue up one component.

My original thought, if we don't care that much about the counter, was to scope the counter to the module (not an instance) and share/increment it across instances, which doesn't require distinguishing with an ID on the chip set. The autogenerated IDs might make a little less sense since they'll strictly increase across instances, but if the user isn't specifying IDs themselves anyway, does it really matter?

Reply:
Updated!